packages/core/src/browser/keybinding.ts

```diff
@@ -492,7 +493,7 @@ export class KeybindingRegistry {
     /**
      * Run the command matching to the given keyboard event.
      */
-    run(event: KeyboardEvent): void {
+    async run(event: KeyboardEvent): Promise<void> {
```

Comment:
I am not sure whether this method can be `async`.

New key events can come while previous are not processed yet. Probable, if we make it async then we should queue events.

cc @svenefftinge